examples/example-phx-1_2/priv/repo/migrations/20180326205407_change_and_convert_page_data_type_in_backups_table.exs

```diff
@@ -0,0 +1,58 @@
+defmodule Example.Repo.Migrations.ChangeAndConvertPageDataTypeInBackupsTable do
+  @moduledoc false
+  use Ecto.Migration
+  alias Thesis.Backup
+  import Thesis.Config
+  import Ecto.Changeset, only: [change: 1, force_change: 3]
+
+  def change do
+    case convert_and_map_page_data() do
+      {:ok, backups} -> execute_migration(backups)
+      _ -> raise "Something went wrong with backup conversion. Please check \
+                  your thesis_backups table and re-run the migration."
+    end
+  end
+
+  def convert_and_map_page_data() do
+    backups = Enum.map(backups(), fn(b) ->
+      %{b | page_data: decompress_page_data(b.page_data)}
+    end)
+
+    {:ok, backups}
+  end
+
+  def execute_migration(backups) do
+    alter table(:thesis_backups) do
+      modify :page_data, :text
+    end
+
+    flush()
+
+    save_converted_page_data(backups)
+  end
+
+  def backups() do
+    repo().all(Backup)
+  end
+
+  def decompress_page_data(page_data) do
+    try do
+      LZString.decompress(page_data)
+    rescue
+      _ -> page_data
+    end
```

Comment:
try and rescue here just in case some data was already migrated before. this probably will never happen unless the server runs out of memory during all of these transactions, the migrations fails, and the migration is re-executed. 

If non-binary is passed into decompress (in the case of already converted data), it'll throw and exception. Then, we just return the json string again.